File: backend/app/schemas/scd.py
from datetime import datetime

from pydantic import BaseModel


class SCDSummary(BaseModel):
    totalRecords: int
    activeRecords: int
    historicalRecords: int
    customersWithMultipleVersions: int
    latestUpdateTimestamp: datetime | None


class CustomerSCDVersion(BaseModel):
    customerId: str
    customerCity: str | None
    customerState: str | None
    effectiveStartDate: datetime
    effectiveEndDate: datetime | None
    isCurrent: bool
    versionNumber: int


class CustomerSCDHistory(BaseModel):
    customerId: str
    versions: list[CustomerSCDVersion]

File: frontend/src/types/scd.ts
export interface SCDSummaryResponse {
  totalRecords: number
  activeRecords: number
  historicalRecords: number
  customersWithMultipleVersions: number
  latestUpdateTimestamp: string | null
}

export interface CustomerSCDVersion {
  customerId: string
  customerCity: string | null
  customerState: string | null
  effectiveStartDate: string
  effectiveEndDate: string | null
  isCurrent: boolean
  versionNumber: number
}

export interface CustomerSCDHistoryResponse {
  customerId: string
  versions: CustomerSCDVersion[]
}
